plugin/trino-teradata/src/test/java/io/trino/plugin/integration/TeradataJdbcConnectorTest.java

```diff
@@ -0,0 +1,942 @@
+ * limitations under the License.
+ */
+
+package io.trino.plugin.integration;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.MoreCollectors;
+import io.trino.Session;
+import io.trino.plugin.integration.clearscape.ClearScapeEnvironmentUtils;
+import io.trino.plugin.jdbc.BaseJdbcConnectorTest;
+import io.trino.plugin.jdbc.JoinOperator;
+import io.trino.spi.connector.JoinCondition;
+import io.trino.sql.planner.plan.AggregationNode;
+import io.trino.sql.planner.plan.FilterNode;
+import io.trino.sql.query.QueryAssertions;
+import io.trino.testing.MaterializedResult;
+import io.trino.testing.MaterializedRow;
+import io.trino.testing.QueryFailedException;
+import io.trino.testing.QueryRunner;
+import io.trino.testing.TestingConnectorBehavior;
+import io.trino.testing.TestingNames;
+import io.trino.testing.assertions.TrinoExceptionAssert;
+import io.trino.testing.sql.SqlExecutor;
+import io.trino.testing.sql.TestTable;
+import org.assertj.core.api.AssertProvider;
+import org.assertj.core.api.Assertions;
+import org.intellij.lang.annotations.Language;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Assumptions;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.OptionalInt;
+import java.util.function.Consumer;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.trino.plugin.teradata.util.TeradataConstants.TERADATA_OBJECT_NAME_LIMIT;
+import static io.trino.testing.TestingConnectorBehavior.SUPPORTS_CREATE_TABLE_WITH_DATA;
+import static io.trino.testing.TestingNames.randomNameSuffix;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Integration test class for Teradata JDBC Connector.
+ * Sets up schema and tables before tests and cleans up afterwards.
+ */
+public class TeradataJdbcConnectorTest
+        extends BaseJdbcConnectorTest
+{
+    protected TestingTeradataServer database;
+
+    public TeradataJdbcConnectorTest()
+    {
+    }
+
+    private static void verifyResultOrFailure(AssertProvider<QueryAssertions.QueryAssert> queryAssertProvider, Consumer<QueryAssertions.QueryAssert> verifyResults, Consumer<TrinoExceptionAssert> verifyFailure)
+    {
+        requireNonNull(verifyResults, "verifyResults is null");
+        requireNonNull(verifyFailure, "verifyFailure is null");
+        QueryAssertions.QueryAssert queryAssert = Assertions.assertThat(queryAssertProvider);
+        verifyResults.accept(queryAssert);
+    }
+
+    @Override
+    protected SqlExecutor onRemoteDatabase()
+    {
+        return database;
+    }
+
+    @Override
+    protected boolean hasBehavior(TestingConnectorBehavior connectorBehavior)
+    {
+        return switch (connectorBehavior) {
+            case SUPPORTS_CREATE_VIEW,
```

Comment:
Sort by alphabetically.

Reply:
addressed in new PR https://github.com/trinodb/trino/pull/26731